apps/admin-page-e2e/src/integration/pages/account.spec.ts

```diff
@@ -0,0 +1,127 @@
+  name: 'telinior the great',
+  phoneNumber: '69420420',
+  streetAddress: 'Y u b like dis?',
+  streetAddress2: 'plds stap',
+  city: 'TestCity',
+  postcode: '6996',
+  country: 'Neverland',
+};
+
+const authMockService = new AuthenticationService();
+
+describe('Account page for admin user', () => {
+  beforeEach(() => {
+    localStorage.setItem(LocalStorageVars.authUser, JSON.stringify(authMockService.getMockUser(AuthUserEnum.authUser)));
+    cy.intercept('GET', '/users/me', {
+      body: mockAdminUser,
+      statusCode: 200,
+    });
+    cy.visit('/account');
+  });
+
+  it('should display the information correctly', () => {
+    cy.get('[data-cy=account-email-input]').should('have.value', 'e2e@test.com');
+    cy.get('[data-cy=account-name-input]').should('have.value', 'telinior the great');
+    cy.get('[data-cy=account-city-input]').should('have.value', 'TestCity');
+    cy.get('[data-cy=account-postcode-input]').should('have.value', '6996');
+    cy.get('[data-cy=account-phone-number-input]').should('have.value', '69420420');
+    cy.get('[data-cy=account-address-input]').should('have.value', 'Y u b like dis?');
+    cy.get('[data-cy=account-address-2-input]').should('have.value', 'plds stap');
+  });
+
+  it('should update the user correctly', () => {
+    cy.intercept('PUT', `/users/${mockAdminUser.userId}`, {
+      statusCode: 200,
+    });
+    cy.get('[data-cy=account-email-input]').clear().type('slap@me.daddy');
+    cy.get('[data-cy=account-name-input]').clear().type('slap daddy');
+    cy.get('[data-cy=account-city-input]').clear().type('slapCity');
+    cy.get('[data-cy=account-postcode-input]').clear().type('0420');
+    cy.get('[data-cy=account-phone-number-input]').clear().type('69696969');
+    cy.get('[data-cy=account-address-input]').clear().type('slap street avenue 69');
+    cy.get('[data-cy=account-address-2-input]').clear().type('420th of course');
+
+    cy.get('[data-cy=update-account-btn]').click();
+  });
+
+  it('should show the correct roles', () => {
+    cy.get('[data-cy=is-user]').should('exist');
+    cy.get('[data-cy=is-developer]').should('exist');
+    cy.get('[data-cy=is-admin]').should('exist');
+    cy.get('[data-cy=is-operation]').should('not.exist');
+  });
+});
+
+describe('change password dialog', () => {
+  beforeEach(() => {
+    localStorage.setItem(LocalStorageVars.authUser, JSON.stringify(authMockService.getMockUser(AuthUserEnum.authUser)));
+    cy.intercept('GET', '/users/me', {
+      body: mockAdminUser,
+      statusCode: 200,
+    });
+    cy.visit('/account');
+    cy.get('[data-cy=change-password-btn]').click();
+  });
+
+  it('should display the change password dialog properly', () => {
+    cy.get('[data-cy=change-password-dialog]').should('exist');
+  });
+
+  it('should have update btn be disabled when password is not good enough', () => {
+    cy.get('[data-cy=update-password-btn]').should('be.disabled');
+    //TODO: check uppercase
```

Comment:
Change the TODO comments into normal ones if they are completed

Also, test for special characters since our regex is wonky about them